pkg/virt-handler/vsock/server.go

```diff
@@ -64,6 +65,13 @@ func (h *Hypervisor) start() {
 }
 
 func (h *Hypervisor) serve() {
+	// Load the vhost_vsock module on demand.
+	if fd, err := os.Open("/dev/vhost-vsock"); err != nil {
```

Comment:
The device node is already present without the module loaded? How is that ensured?

Reply:
Same as `vhost-net`, the module has to be compiled in the OS. Then the device will be present. These modules are loaded on demand by udev (device manager).

If the device doesn't present, then the entire VSOCK functionality is broken.

Reply:
I agree with @zhuchenwang that the vsock functionality would not work if the device is not there. Right now it only gets activated if the  vsock feature gate is enabled and there we just assume that the device is there.

I think we could in a follow-up improve the system service startup by, like the device-plugin, having a file-watcher.

The original behaviour which I added for the system-service is to just blindly periodically try to start the system service if the vsock feature gate is enabled. I can see that my "naive" approach may potentially pollute logs quite a bit on heterogenous node setups.

This PR does not improve that part of the system-service startup issue, but should definitely fix loading the right module.